dev/com.ibm.ws.microprofile.config_fat/fat/src/com/ibm/ws/microprofile/config/fat/tests/SimultaneousRequestsTest.java

```diff
@@ -0,0 +1,107 @@
+
+import java.io.File;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ExecutorService;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.ws.microprofile.appConfig.ordForDefaults.test.OrdinalsForDefaultsTestServlet;
+import com.ibm.ws.microprofile.config.fat.suite.SharedShrinkWrapApps;
+
+import componenttest.annotation.Server;
+import componenttest.annotation.TestServlet;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+import com.ibm.ws.fat.util.browser.WebBrowser;
+import com.ibm.ws.fat.util.browser.WebBrowserException;
+import com.ibm.ws.fat.util.browser.WebBrowserFactory;
+import com.ibm.ws.fat.util.browser.WebResponse;
+import com.ibm.ws.microprofile.config.fat.suite.RepeatConfig11EE7;
+import com.ibm.ws.microprofile.config.fat.suite.RepeatConfig12EE8;
+
+import com.ibm.websphere.simplicity.log.Log;
+
+@RunWith(FATRunner.class)
+public class SimultaneousRequestsTest extends FATServletClient {
+
+    public static final String APP_NAME = "simultaneousRequests";
+
+    @Server("SimultaneousRequestsServer")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+
+        WebArchive simultaneousRequests_war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                        .addPackages(true, "com.ibm.ws.microprofile.appConfig.simultaneousRequests.test")
+                        .addAsLibrary(SharedShrinkWrapApps.getTestAppUtilsJar())
+                        .addAsManifestResource(new File("test-applications/" + APP_NAME + ".war/resources/META-INF/permissions.xml"), "permissions.xml")
+                        .addAsManifestResource(new File("test-applications/" + APP_NAME + ".war/resources/META-INF/microprofile-config.properties"),
+                                               "microprofile-config.properties");
+
+        ShrinkHelper.exportDropinAppToServer(server, simultaneousRequests_war);
+
+        server.startServer();
+    }
+
+    @Test
+    public void testSimultaneousRequests() throws Exception {
+
+        Callable<WebResponse> callableWebResponse = new Callable<WebResponse>() {
+            @Override
+            public WebResponse call() {
+                try {
+                    WebResponse wr = WebBrowserFactory.getInstance().createWebBrowser().request("http://" + server.getHostname() + ":" + server.getHttpDefaultPort() + "/simultaneousRequests/");
+                    return wr;
+                } catch (WebBrowserException e) {
```

Comment:
We could just let this exception be thrown, then we'd get a better error message if a request fails. Callables are allowed to declare exceptions.